docs/user/vsphere/install_upi.md

```diff
@@ -0,0 +1,380 @@
+
+* etcd
+
+    For each control plane machine, OpenShift 4.x requires DNS records `etcd-$idx.$cluster_name.$base_domain` to point to `$idx`'th control plane machine. The DNS record must resolve to an unicast IPV4 address for the control plane machine and the records must be resolvable from all the nodes in the cluster.
+
+    For each control plane machine, OpenShift 4.x also requires a SRV DNS record for etcd server on that machine with priority `0`, weight `10` and port `2380`. For 3 control plane cluster, the records look like:
+
+    ```plain
+    # _service._proto.name.                            TTL   class SRV priority weight port target.
+    _etcd-server-ssl._tcp.$cluster_name.$base_domain   86400 IN    SRV 0        10     2380 etcd-0.$cluster_name.$base_domain.
+    _etcd-server-ssl._tcp.$cluster_name.$base_domain   86400 IN    SRV 0        10     2380 etcd-1.$cluster_name.$base_domain.
+    _etcd-server-ssl._tcp.$cluster_name.$base_domain   86400 IN    SRV 0        10     2380 etcd-2.$cluster_name.$base_domain.
+    ```
+
+* OpenShift Routes
+
+    OpenShift 4.x requires the DNS record `*.apps.$cluster_name.$base_domain` to point to the Load balancer targeting the machines running the ingress router pods. This record must be resolvable by both clients external to the cluster and from all the nodes within the cluster.
+
+## Getting ignition configs for machines
+
+The OpenShift Installer provides administrators various assets that are required to create an OpenShift cluster, namely:
+
+* Ignition configs: The OpenShift Installer provides Ignition configs that should be used to configure the RHCOS based bootstrap and control plane machines using `bootstrap.ign`  and `master.ign` respectively. The OpenShift Installer also provides `worker.ign` that can be used to configure the RHCOS based `worker` machines, but also can be used as source for configuring RHEL based machines [todo-link-to-BYO-RHEL].
+
+* Admin Kubeconfig: The OpenShift Installer provides a kubeconfig with admin level privileges to Kubernetes APIServer.
+
+    NOTE: This kubeconfig is configured to use `api.$cluster_name.$base_domain` DNS name to communicate with the Kubernetes APIServer.
+
+### Setting up install-config for installer
+
+The OpenShift installer uses an [Install Config][install-config] to drive all install time configuration.
+
+An example install config for vSphere UPI is as follows:
+
+```yaml
+apiVersion: v1beta4
+## The base domain of the cluster. All DNS records will be sub-domains of this base and will also include the cluster name.
+baseDomain: example.com
+compute:
+- name: worker
+  replicas: 1
+controlPlane:
+  name: master
+  replicas: 3
+metadata:
+  ## The name for the cluster
+  name: test
+platform:
```

Comment:
TODO:  this can be updated to follow the new platform. PR in flight https://github.com/openshift/installer/pull/1591